### ISSUE TITLE
FlutterSecureStorage on macos build macOS (darwin) fails

### ISSUE THREAD

**Issue by ysimonx** [mogol/flutter_secure_storage#689]
Tested on 2 différent macbook m1

```
final _storage = const FlutterSecureStorage();

await _storage.write(key: "lastStatus", value: "chrono");
String? x = await _storage.read(key: "lastStatus");
print(x);

```           
the value of "x" is not "chrono" ... don't know why it just occured since last week

```

flutter_secure_storage:
    version: "9.0.0"
  flutter_secure_storage_linux:
    version: "1.2.0"
  flutter_secure_storage_macos:
    version: "3.0.1"
  flutter_secure_storage_platform_interface:
    version: "1.0.2"
  flutter_secure_storage_web:
    version: "1.1.2"
  flutter_secure_storage_windows:
    version: "3.0.0"
         
```

```

[✓] Flutter (Channel stable, 3.16.0, on macOS 14.2.1 23C71 darwin-arm64, locale fr-FR)
    • Flutter version 3.16.0 on channel stable at /Users/ysimonx/Developpement/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision db7ef5bf9f (4 months ago), 2023-11-15 11:25:44 -0800
    • Engine revision 74d16627b9
    • Dart version 3.2.0
    • DevTools version 2.28.2

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /Users/ysimonx/Library/Android/sdk
    • Platform android-33, build-tools 33.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15E204a
    • CocoaPods version 1.11.3

```

**Comment by gnunicorn** [comment]
Hey @ysimonx , there is a patch for Apple systems in main that hasn't been released yet to  pub.dev: https://github.com/mogol/flutter_secure_storage/pull/629 . Could try testing against main and report if that fixes it for you?

**Comment by QCIPaulCardno** [comment]
@ysimonx I had a similar issue with updating the key value not working. If you delete the key before you write to it it will work. Not a nice work around but it does get it going for me.
try:
await _storage.delete(key: "lastStatus");
await _storage.write(key: "lastStatus", value: "chrono");
String? x = await _storage.read(key: "lastStatus");
print(x);

**Comment by juliansteenbakker** [comment]
I have released v9.1.0 which should fix this issue.
